phys/module_surface_driver.F

```diff
@@ -2920,13 +3061,37 @@ SUBROUTINE surface_driver(                                         &
        ENDDO                                                  !urban
      ENDIF
 
+ ! TEB
+      IF(SF_URBAN_PHYSICS.eq.4) THEN
+        DO j=j_start(ij),j_end(ij)                             !urban
+          DO i=i_start(ij),i_end(ij)                           !urban
+            IF( IVGTYP(I,J) == ISURBAN .or. IVGTYP(I,J) == LOW_DENSITY_RESIDENTIAL .or. &  
+              IVGTYP(I,J) == HIGH_DENSITY_RESIDENTIAL .or. IVGTYP(I,J) == HIGH_INTENSITY_INDUSTRIAL ) THEN 
+              ! TH2_URB2D(I,J): temperature urb (not potential temp)   
+              T2(I,J)   = TH2_URB2D(i,j)
+              TH2(I,J)  = T2(I,J)*((1.E5/PSFC(I,J))**RCP)
+              Q2(I,J)   = Q2_URB2D(i,j)
+              U10(I,J)  = U10_URB2D(I,J)                       !urban
+              V10(I,J)  = V10_URB2D(I,J)                       !urban
+            END IF
+          ENDDO
+        ENDDO
+      END IF
+ !end TEB
+!
+
 !------------------------------------------------------------------
 
        ELSE
          CALL wrf_error_fatal('Lacking arguments for LSM in surface driver')
        ENDIF
 
      CASE (NOAHMPSCHEME)
+
```

Comment:
@dmey 
This type of test is OK, but should also be in the share/module_check_a_mundo.F file. The testing here helps users that are looking at code to understand the logic and restrictions. The test in check_a_mundo is the proper place to determine inconsistent namelist settings.